agimus_controller/tests/test_ocp_croco_joint_state.py

```diff
@@ -0,0 +1,166 @@
+import unittest
+import numpy as np
+from pathlib import Path
+import example_robot_data as robex
+import pinocchio as pin
+from agimus_controller.ocp.ocp_croco_joint_state import OCPCrocoJointState
+from agimus_controller.ocp_param_base import OCPParamsBaseCroco
+from agimus_controller.factory.robot_model import RobotModels, RobotModelParameters
+from agimus_controller.trajectory import (
+    TrajectoryPoint,
+    TrajectoryPointWeights,
+    WeightedTrajectoryPoint,
+)
+
+
+class TestOCPWarmstart(unittest.TestCase):
+    def setUp(self):
+        # Initialize robot
+        robot = robex.load("panda")
+        urdf_path = Path(robot.urdf)
+        srdf_path = Path(robot.urdf.replace("urdf", "srdf"))
+        urdf_meshes_dir = urdf_path.parent.parent.parent.parent.parent
+        free_flyer = False
+        locked_joint_names = []
+        reduced_nq = robot.model.nq - len(locked_joint_names)
+        full_q0 = np.zeros(robot.model.nq)
+        q0 = np.zeros(reduced_nq)
+        armature = np.full(reduced_nq, 0.1)
+
+        # Store shared initial parameters
+        self.params = RobotModelParameters(
+            q0=q0,
+            full_q0=full_q0,
+            free_flyer=free_flyer,
+            locked_joint_names=locked_joint_names,
+            urdf_path=urdf_path,
+            srdf_path=srdf_path,
+            urdf_meshes_dir=urdf_meshes_dir,
+            collision_as_capsule=True,
+            self_collision=False,
+            armature=armature,
+        )
+
+        self.robot_models = RobotModels(self.params)
+        self.robot_model = self.robot_models.robot_model
+        self.robot_data = self.robot_model.createData()
+
+        # OCP parameters
+        dt = 0.05
+        self._horizon_size = 30
+        solver_iters = 100
+        callbacks = False
+
+        self._ocp_params = OCPParamsBaseCroco(
+            dt=dt,
+            horizon_size=self._horizon_size,
+            solver_iters=solver_iters,
+            callbacks=callbacks,
+            qp_iters=500,
+        )
+
+    def test_ocp_solution(self):
+        # Set initial state
+        q0 = pin.neutral(self.robot_model)
+
+        # Generate warmstart trajectories
+        amplitude = np.deg2rad(20)
+        frequency = 2
```

Comment:
```suggestion
        frequency = 2.0
```
Just to make it `float`